### DISCUSSION TITLE
how exclude changes in changelog by some "service account" job ?

### DISCUSSION THREAD

**Discussion by jancovic** [netbox-community/netbox#12071] [General]
Hi

I am using Ricardo's Vcenter sync  plugin  to netbox  https://github.com/bb-Ricardo/netbox-sync  to sync VMs to netbox.
Problem is that there are hundreds of changes every day ( daily scan ).
Job is running with separate service user, but still is filling all changelog and important changes on physical devices by real users are lost there.
Is there any option how to exclude this logs ?  Also for capacity reasons will be good I think.
We have now 1 month retention setup.

**Comment by mtinberg** [comment]
Unless netbox-sync is creating no-op changelog records when nothing has changed, which could be fixed in the script by having it check each field for differences before submitting an update if it doesn't already do that, scripts and API calls do make changelog entries when they update records, so there is no way around this.

You can exclude changes from view based on userid though, to help highlight the by-hand changes from automated ones, eg /extras/changelog/?user_id__n=<id of API/automation user>

I have similar issues and currently have retention set to 500 and some days with 2,500,000 or so records without issue, of which apparently about 15,000 are made by people and not the automation.  Performance of changelog viewing and filtering has been fine.

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: AnderAbc ***@***.***>
Sent: Monday, March 27, 2023 11:41 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] how exclude changes in changelog by some "service account" job ? (Discussion #12071)


Hi

I am using Ricardo's Vcenter sync plugin to netbox https://github.com/bb-Ricardo/netbox-sync to sync VMs to netbox.
Problem is that there are hundreds of changes every day ( daily scan ).
Job is running with separate service user, but still is filling all changelog and important changes on physical devices by real users are lost there.
Is there any option how to exclude this logs ? Also for capacity reasons will be good I think.
We have now 1 month retention setup.

**Comment by mtinberg** [comment]
Also for capacity reasons will be good I think.
We have now 1 month retention setup.
Just to check, my PostgreSQL DB is about 5GB with around 7,000 Prefix, 40,000 IPAddress, 6,000 VLANs, 1,000 Sites, 20,000 Devices, 300,000 Interfaces plus the 2,500,000 ChangeLogs mentioned, which is still small enough to fit in RAM on a Smartphone so performance and capacity hasn't been an issue.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison
